Automattic/pocket-casts-android
# Listening history - Add local search

## Description
This adds local search to listening history

## Testing Instructions
1. Login with an account with sufficient listening history
2. Go to `Profile` tab -> `Listening History`
3. Search for an episode that exists in listening history
4. ✅ Notice that the episode appears in the results if the search term exists in episode title or podcast title
5. Search for an episode that does not exists in listening history
6. ✅ Notice that empty results screen is displayed

## Screenshots or Screencast 

https://github.com/user-attachments/assets/c33c94ee-3ad6-4d56-b12d-80e44350d465


## Checklist
 
#### I have tested any UI changes...
